chromium_src/ios/chrome/browser/net/ios_chrome_network_delegate.cc

```diff
@@ -0,0 +1,41 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "ios/chrome/browser/net/ios_chrome_network_delegate.h"
+
+#define IOSChromeNetworkDelegate IOSChromeNetworkDelegate_ChromiumImpl
+#include "src/ios/chrome/browser/net/ios_chrome_network_delegate.cc"
+#undef IOSChromeNetworkDelegate
+
+#include "brave/common/network_constants.h"
+#include "extensions/common/url_pattern.h"
+
+namespace {
+
+void AddBraveServicesKeyHeader(net::URLRequest* request) {
+  static URLPattern brave_proxy_pattern(URLPattern::SCHEME_HTTPS,
```

Comment:
I'm a little worried that this could potentially get out of sync with desktop if we add any other domains. Can we make a single constant somewhere that combines both and share it with both?

Reply:
desktop uses `network::ResourceRequest` instead of `net::URLRequest` (which means different APIs to set the header and get the URL): https://github.com/brave/brave-core/blob/1cb5818aa0b70666c6aeea5ea9c06cc4e712171a/browser/net/brave_system_request_handler.cc#L22 but I suppose we could refactor this instead to be something like and then do the actual header alteration and URL extraction at their respective implementation points

```cpp
bool ShouldAddBraveServicesKeyHeader(const GURL& url) {
   ...
}
```

Reply:
Where would you consider a good place to drop this shared code? Somewhere in `//brave/net` perhaps?

Reply:
technically `//brave/net` is the equivalent of `//net` so it really shouldn't have Brave specific stuff in it so I would just stick it in `//brave/common` or maybe `//brave/common/net`